test/unit/JobsManager.js

```diff
@@ -40,176 +47,288 @@ contract("JobsManager", accounts => {
+
+            await jobsManager.deposit({from: broadcaster, value: 1000})
             // Broadcaster creates job 0
-            await jobsManager.job(streamId, transcodingOptions, maxPricePerSegment, endBlock0, {from: broadcaster})
-            const endBlock1 = 100 + 20
+            await jobsManager.job("foo", transcodingOptions, 1, currentBlock + 20, {from: broadcaster})
             // Broadcaster creates job 1
-            await jobsManager.job(streamId, transcodingOptions, maxPricePerSegment, endBlock1, {from: broadcaster})
-
-            // Job 1 ends
-            await fixture.roundsManager.mineBlocks(20)
+            await jobsManager.job("foo", transcodingOptions, 1, currentBlock + 50, {from: broadcaster})
         })
 
-        it("should fail for invalid job id", async () => {
+        it("should fail if the job does not exist", async () => {
             const invalidJobId = 2
-            // Transcoder claims work for invalid job id
-            await expectThrow(jobsManager.claimWork(invalidJobId, segmentRange, claimRoot, {from: electedTranscoder}))
+            await expectThrow(jobsManager.claimWork(invalidJobId, segmentRange, claimRoot, {from: transcoder}))
+        })
+
+        it("should fail if the job is inactive", async () => {
+            await fixture.roundsManager.setMockUint256(functionSig("blockNum()"), currentBlock + 20)
+            const inactiveJobId = 0
+            await expectThrow(jobsManager.claimWork(inactiveJobId, segmentRange, claimRoot, {from: transcoder}))
+        })
+
+        it("should fail if segment range is invalid", async () => {
+            await expectThrow(jobsManager.claimWork(1, [3, 0], claimRoot, {from: transcoder}))
+        })
+
+        it("should fail if the transcoder address is set and does not match caller address", async () => {
+            // First claim - set transcoder address
+            await jobsManager.claimWork(1, segmentRange, claimRoot, {from: transcoder})
+
+            await expectThrow(jobsManager.claimWork(1, segmentRange, claimRoot, {from: accounts[2]}))
         })
 
-        it("should fail for inactive job", async () => {
-            const inactiveJobId = 1
-            // Transcoder claims work for inactive job
-            await expectThrow(jobsManager.claimWork(inactiveJobId, segmentRange, claimRoot, {from: electedTranscoder}))
+        it("should fail if the transcoder address is not set and the caller is not assigned", async () => {
+            await expectThrow(jobsManager.claimWork(1, segmentRange, claimRoot, {from: accounts[2]}))
         })
 
-        it("should fail if the transcoder is assigned and the sender is not the assigned transcoder", async () => {
-            await jobsManager.claimWork(jobId, segmentRange, claimRoot, {from: electedTranscoder})
+        it("should set the transcoder address if this is the first claim and the caller is assigned", async () => {
```

Comment:
Do unit tests in truffle have some kind of guaranteed ordering?  Feels like we should check for `jInfo[4] == 0`, and then `claimWork`, and see that `jInfo[4]` changed.

Reply:
The beforeEach hook for this describe block ensures that each test starts with the same exact state so that test ordering does not matter so `jInfo[4]` will always be 0 before calling `claimWork` and the tests for `job` test that `jInfo[4]` is initialized to be the null address.